### PR TITLE
feat: add graasp branding and show all platforms

### LINKED ISSUE #448: Redesign login with better branding

### PR DESCRIPTION
**Pull request by pyphilia**
To have something simpler, I chose to show all platforms at once. I've added a "Graasp" branding at the top so it's clear it's within the Graasp domain.

Screenshot 2024-09-12 at 14 47 10

close #448

### CODE REVIEW

File: cypress/e2e/SignUp.cy.ts [hunk 1]
```diff
@@ -25,7 +25,10 @@ describe('SignUp', () => {
     it('Cannot Sign Up Without Accepting Terms', () => {
       cy.get(`#${SIGN_UP_BUTTON_ID}`).should('be.disabled');
     });
-    it('Sign Up Is Available When Accepting All Terms', () => {
+    it.only('Sign Up Is Available When Accepting All Terms', () => {
```
Comment (ReidyT) [inline cypress/e2e/SignUp.cy.ts:28]:
```suggestion
    it('Sign Up Is Available When Accepting All Terms', () => {
```

File: src/components/BrandingLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { GraaspLogo } from '@graasp/ui';
+
+import { Stack, Typography, useTheme } from '@mui/material';
+
+export const BrandingLogo = () => {
+  const theme = useTheme();
+
+  return (
+    <div style={{ position: 'absolute', top: 0, left: 0 }}>
```
Comment (ReidyT) [inline src/components/BrandingLogo.tsx:9]: I would increase the position a bit

File: src/components/LeftContentContainer.tsx [hunk 3]
```diff
@@ -1,67 +1,93 @@
...
-        }}
-        width="100%"
-        justifyContent="center"
-        alignItems="center"
-        py={4}
+        direction="row"
+        margin="auto"
+        minHeight="100svh"
         sx={{
           backgroundImage: `url("data:image/svg+xml,%3Csvg width='100' height='20' viewBox='0 0 100 20' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M21.184 20c.357-.13.72-.264 1.088-.402l1.768-.661C33.64 15.347 39.647 14 50 14c10.271 0 15.362 1.222 24.629 4.928.955.383 1.869.74 2.75 1.072h6.225c-2.51-.73-5.139-1.691-8.233-2.928C65.888 13.278 60.562 12 50 12c-10.626 0-16.855 1.397-26.66 5.063l-1.767.662c-2.475.923-4.66 1.674-6.724 2.275h6.335zm0-20C13.258 2.892 8.077 4 0 4V2c5.744 0 9.951-.574 14.85-2h6.334zM77.38 0C85.239 2.966 90.502 4 100 4V2c-6.842 0-11.386-.542-16.396-2h-6.225zM0 14c8.44 0 13.718-1.21 22.272-4.402l1.768-.661C33.64 5.347 39.647 4 50 4c10.271 0 15.362 1.222 24.629 4.928C84.112 12.722 89.438 14 100 14v-2c-10.271 0-15.362-1.222-24.629-4.928C65.888 3.278 60.562 2 50 2 39.374 2 33.145 3.397 23.34 7.063l-1.767.662C13.223 10.84 8.163 12 0 12v2z' fill='%23f2f2ff' fill-opacity='0.82' fill-rule='evenodd'/%3E%3C/svg%3E")`,
```
Comment (ReidyT) [inline src/components/LeftContentContainer.tsx:27]: I know that it was not a change from this PR, but we should extract this in a const!

File: src/components/SignUp.tsx [hunk 4]
```diff
@@ -214,7 +214,7 @@ const SignUp = () => {
             loading={isLoadingSignUp || isLoadingMobileSignUp}
             onClick={handleRegister}
             fullWidth
-            disabled={!userHasAcceptedAllTerms}
+            disabled={!userHasAcceptedAllTerms || !email.length || !name.length}
```
Comment (ReidyT) [inline src/components/SignUp.tsx:217]: 🥳

File: src/langs/de.json [hunk 5]
```diff
@@ -1,12 +1,11 @@
 {
   "BACK_BUTTON": "Zurück",
-  "EMAIL_FIELD_TEXT": "Email",
   "EMAIL_SIGN_IN_METHOD": "E-Mail Anmeldung",
   "MOBILE_APP_NOT_INSTALLED_MESSAGE": "Sie haben einen mobilen Link angefordert, aber unsere App ist auf diesem Gerät nicht installiert.",
   "MOBILE_BACK_TO_LOGIN": "Zurück zur Anmeldung",
   "MOBILE_GET_APP_FROM_APPLE_STORE": "Im Apple App Store herunterladen",
   "MOBILE_GET_APP_FROM_GOOGLE_PLAY_STORE": "Im Google Play Store herunterladen",
-  "NAME_FIELD_LABEL": "Name",
+  "NAME_FIELD_LABEL": "Name*",
```
Comment (ReidyT) [inline src/langs/de.json:8]: I thought we should remove the `*`?

**Review by ReidyT — CHANGES_REQUESTED** [review]
Thanks for this PR! I really like it 🔥. I tried to reverse some of them, but finally I think it's better the way you did.

image

I would just add more top and left margin to the branding. Also, on mobile, I'm not a big fan of having the branding and the Graasp icon above the login. Maybe we could merge the two on mobile?

image

Also, it's possible I misunderstood, but I thought we should avoid the `*` in the placeholder, shouldn't we?

File: src/components/LeftContentContainer.tsx [hunk 6]
```diff
@@ -1,67 +1,93 @@
...
+          justifyContent="center"
+          alignItems="center"
+        >
+          <APIChecker />
+          <Stack spacing={3}>
+            <PlatformContent
+              Icon={BuildIcon}
+              text={t(AUTH.BUILDER_BACKGROUND_TEXT)}
+              name={t(AUTH.BUILDER_BACKGROUND_TEXT_PLATFORM)}
+              color="#1ECAA5"
```
Comment (spaenleh) [inline src/components/LeftContentContainer.tsx:46]: These are defined in the `ACCENT_COLORS` in ui.

**Pull request merged**